### DISCUSSION TITLE
`Unsafe.As` vs `Unsafe.BitCast`

### DISCUSSION THREAD

**Discussion by colejohnson66** [dotnet/runtime#104626] [General] [answered]
What's the difference between `Unsafe.As` and `Unsafe.BitCast`? I see that `Unsafe.BitCast` is restricted to `struct` and doesn't require a `ref`, but they both do the same thing: reinterpret the bits of the target. What's the practical difference?

**Comment by tannergooding** [comment, marked as answer]
`Unsafe.BitCast` works on value types and doesn't require taking an address, which can improve the overall codegen/perf in some complex scenarios (in simple ones the JIT will normally recognize its not actually "address-taken").

It also helps guarantee correctness as it also requires the input/output be the same size and therefore that you aren't accessing bits that don't exist.
